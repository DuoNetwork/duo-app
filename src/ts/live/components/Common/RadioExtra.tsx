//import moment from 'moment';
import { Radio } from 'antd';
import * as React from 'react';
import { SCardRadioExtraDiv, SRadioGroup } from '../Cards/_styled';

const RadioButton = Radio.Button;

interface IProps {
	text?: string;
	onChange: () => void;
	left: string,
	right: string,
	isLeft: boolean
}

export default class RadioExtra extends React.PureComponent<IProps> {
	public render() {
		const {text, onChange, left, right, isLeft } = this.props;
		return (
			<SCardRadioExtraDiv>
				<div className="extend-extra-wrapper">
					{text ? <div className="tag-title">Fee in</div> : null}
					<SRadioGroup
						defaultValue={left}
						size="small"
						onChange={onChange}
						value={isLeft ? left : right}
					>
						<RadioButton value={left}>{left}</RadioButton>
						<RadioButton value={right}>{right}</RadioButton>
					</SRadioGroup>
				</div>
			</SCardRadioExtraDiv>
		);
	}
}
