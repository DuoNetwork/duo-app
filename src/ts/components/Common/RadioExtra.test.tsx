import { shallow } from 'enzyme';
import * as React from 'react';
// import { IConversion } from 'ts/common/types';
import RadioExtra from './RadioExtra';

describe('AuthCard Test', () => {
	describe('User Login', () => {
		const text = 'test';
		const onChange = jest.fn();
		const left = 'test';
		const right = 'test';
		const isLeft = true;
		const rightPadding = true;

		it('Test Snapshot', () => {
			const wrapper1 = shallow(<RadioExtra text={text} onChange={onChange} left={left} right={right} isLeft={isLeft} rightPadding={rightPadding} />);
			expect(wrapper1).toMatchSnapshot();
		});
	});
});
