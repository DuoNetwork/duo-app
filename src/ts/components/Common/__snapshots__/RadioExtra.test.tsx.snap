// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthCard Test User Login Test Snapshot 1`] = `
<styled.div>
  <div
    className="extend-extra-wrapper true"
  >
    <div
      className="tag-title"
    >
      Fee in
    </div>
    <Styled(RadioGroup)
      defaultValue="test"
      onChange={[MockFunction]}
      size="small"
      value="test"
    >
      <RadioButton
        prefixCls="ant-radio-button"
        value="test"
      >
        test
      </RadioButton>
      <RadioButton
        prefixCls="ant-radio-button"
        value="test"
      >
        test
      </RadioButton>
    </Styled(RadioGroup)>
  </div>
</styled.div>
`;
